substrate/frame/support/procedural/src/lib.rs

```diff
@@ -1061,27 +1058,6 @@ pub fn disable_frame_system_supertrait_check(_: TokenStream, _: TokenStream) ->
 	pallet_macro_stub()
 }
 
-/// To generate a `Store` trait associating all storages, annotate your `Pallet` struct with
-/// the attribute `#[pallet::generate_store($vis trait Store)]`, e.g.:
-///
-/// ```ignore
-/// #[pallet::pallet]
-/// #[pallet::generate_store(pub(super) trait Store)]
-/// pub struct Pallet<T>(_);
-/// ```
-/// More precisely, the `Store` trait contains an associated type for each storage. It is
-/// implemented for `Pallet` allowing access to the storage from pallet struct.
-///
-/// Thus when defining a storage named `Foo`, it can later be accessed from `Pallet` using
-/// `<Pallet as Store>::Foo`.
-///
-/// NOTE: this attribute is only valid when applied _directly_ to your `Pallet` struct
-/// definition.
-#[proc_macro_attribute]
-pub fn generate_store(_: TokenStream, _: TokenStream) -> TokenStream {
-	pallet_macro_stub()
```

Comment:
I dont think we ever issued a deprecation warning.  
We need to do that first and wait for at least 6 months for downstream to adapt.

Reply:
@philoniare of a downstream project use trait Store today, will they have a compiler warning? 

If yes, then we can merge this. 

If no, then Oliver is right and we need to wait. 

I hope it is the former.

Reply:
@kianenigma yep, as of today, if a pallet developer uses `Store` trait, they're getting the following compiler warning:
```
warning: use of deprecated struct `pallet::_::Store`: 
                 Use of `#[pallet::generate_store(pub(super) trait Store)]` will be removed after July 2023.
                 Check https://github.com/paritytech/substrate/pull/13535 for more details.
  --> src/lib.rs:27:3
   |
27 |     #[pallet::generate_store(trait Store)]
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

```